syntax="proto3";
package eos.console;

message RouteProto {
  message LinkProto {
    message Endpoint {
      string fqdn      = 1;
      uint32 xrd_port  = 2;
      uint32 http_port = 3;
   }

   string path = 1;
   repeated Endpoint endpoints = 2;
  }

  message UnlinkProto {
    string path = 1;
  }

  message ListProto {
    string path = 1;
  }

  oneof subcmd {
    ListProto list     = 1;
    LinkProto link     = 2;
    UnlinkProto unlink = 3;
  }
}
